xarray/plot/plot.py

```diff
@@ -433,11 +470,16 @@ def newplotfunc(darray, x=None, y=None, ax=None, row=None, col=None,
         # This allows the user to pass in a custom norm coming via kwargs
         kwargs.setdefault('norm', cmap_params['norm'])
 
-        ax, primitive = plotfunc(xval, yval, zval, ax=ax,
-                                 cmap=cmap_params['cmap'],
-                                 vmin=cmap_params['vmin'],
-                                 vmax=cmap_params['vmax'],
-                                 **kwargs)
+        if 'imshow' == plotfunc.__name__ and isinstance(aspect, basestring):
```

Comment:
I think it would be better to avoid this special case, and simply not pass `aspect` on to `imshow`. Xarray `imshow` is already overloaded to display in data coordinates (not pixels) so I think this is pretty reasonable.